fix: Website Enhancements

netchampfaris opened the pull request:
## New Features

#### 1. Section with Collapsible Content

<details>
<summary>Screenshot</summary>


![image](https://user-images.githubusercontent.com/9355208/90518242-1efc3800-e184-11ea-8d4d-eef1778fdc90.png)


</details>



#### 3. Footer

Updated styling for grouped links.

<details>
<summary>Screenshot</summary>


![image](https://user-images.githubusercontent.com/9355208/90392306-4089f080-e0ac-11ea-91e5-a88d21503712.png)


</details>

#### 4. Section with Features

<details>
<summary>Screenshot</summary>


![image](https://user-images.githubusercontent.com/9355208/90518006-cf1d7100-e183-11ea-9545-30b817346fbe.png)


</details>

#### 5. Section with Embed


<details>
<summary>Screenshot</summary>


![image](https://user-images.githubusercontent.com/9355208/90518190-0d1a9500-e184-11ea-8053-45626da2e48c.png)


</details>

#### 6. CSS Variables

Common CSS values like colors and font sizes are now available to use via CSS Custom Properties.

**Usage Example:**
```html
<div style="border: 1px solid var(--gray-400); font-size: var(--font-size-lg)">Content</div>
```

#### 7. Customizable Navbar and Footer

The website Navbar and Footer can now be customized using custom Web Templates. Just create a Web Template and set it's type to Navbar/Footer and change the Navbar/Footer template in Website Settings.

### Extended spacing scale

Bootstrap provides a spacing scale of 1 to 5. So you can use classes like `mt-2`, `mt-5` to apply a margin top of 0.5rem and 3rem respectively. This PR extends the spacing scale from 1 through 64.

<details>
<summary>Spacing scale</summary>


```
$spacers: (
	0: 0,
	1: 0.25rem,
	2: 0.5rem,
	3: 0.75rem,
	4: 1rem,
	5: 1.25rem,
	6: 1.5rem,
	8: 2rem,
	10: 2.5rem,
	12: 3rem,
	14: 3.5rem,
	16: 4rem,
	18: 4.5rem,
	20: 5rem,
	22: 5.5rem,
	24: 6rem,
	28: 7rem,
	32: 8rem,
	36: 9rem,
	40: 10rem,
	44: 11rem,
	48: 12rem,
	52: 13rem,
	56: 14rem,
	64: 16rem,
);
```


</details>



## Fixes to existing features

1. Hero image layout reflow on lazy load

   In Hero with Right Image, the image would cause a "jump" in the page when it is loaded. This is now fixed.

2. Hero now supports center alignment.
3. Section with Image supports center alignment.
4. Standard Web Templates will not be editable in production mode
5. Support style tags in Web Template
   Style tags will now be extracted and loaded once per Web Template. So, if the template is used multiple times on the same page, the style tag will be rendered only once.
6. `web_block` helper for rendering web templates
7. Markdown template supports center alignment.
8. Render html as text in HTMLEditor to prevent html injection
9. Split section with Image
   - Vertical align style for content
   - Configurable column split

netchampfaris committed:
feat: Section with Collapsible Content

netchampfaris committed:
feat: Section with Image align center

netchampfaris committed:
feat: Footer

- Split footer in files
- Footer grouped links styling
- Put footer logo and extension in one row
- Delete unused footer_items.html
- Uncheck Right when adding Footer Items in Website Settings

netchampfaris committed:
fix: Default styling for h3

netchampfaris committed:
fix: overflow auto in grid field

netchampfaris committed:
fix: Remove margin bottom of last child in md

netchampfaris committed:
fix: Add template validation in web_blocks

netchampfaris committed:
fix: Syntax highlighting for Custom SCSS field

netchampfaris committed:
fix: Fields is not mandatory in Web Template

netchampfaris committed:
feat: Support for Table field via Table Break in Web Template

netchampfaris committed:
fix: empty init file

netchampfaris committed:
fix: revert overflow auto

netchampfaris committed:
fix: Apply vertical margin only in sidebar layout

netchampfaris committed:
fix: Hero image layout reflow on lazy load

Before this change the hero image would cause a layout reflow resulting
a layout jump when the image is loaded

netchampfaris committed:
feat: Section with Features

netchampfaris committed:
feat: Hero align center style

netchampfaris committed:
fix: Footer styles

netchampfaris committed:
fix: Collapsible icon

netchampfaris committed:
fix: Responsive styling for Section with Features

netchampfaris committed:
feat: Section with Embed

netchampfaris committed:
fix: Set common css variables on root

netchampfaris committed:
fix: Conditionally render subtitle

netchampfaris committed:
feat: Customizable navbar and footer with Web Template

netchampfaris committed:
fix: Show Edit DocType button in Singles Form in dev mode

netchampfaris committed:
fix: import

netchampfaris committed:
feat: Extended spacing scale

ghost commented:
✅ Errors or warnings were resolved via the commit 958d5f5! Please visit [Sider](https://sider.review/gh/repos/1864194/pulls/11285) for details.

---

You can [turn off this notification](https://sider.review/gh/repos/1864194/settings#notifications) if you don't want to receive it from now on.

netchampfaris committed:
fix: Ignore SVG images

netchampfaris committed:
feat: Slideshow Web Template

- Clickable slideshow items

netchampfaris committed:
fix: Set og:type to website for web page

dottenbr commented:
Awesome!

Commit:
fix: Set _context_dict always

netchampfaris committed:
fix: Indicator check

netchampfaris committed:
fix: Web Template

- Disable standard web template form in production
- Prevent standard web template creation in production
- Set type for all templates

netchampfaris committed:
fix(Small CTA: Hide subtitle and cta if not set

netchampfaris committed:
fix: Print error to avoid silent failure

netchampfaris committed:
feat: Support style tags in Web Template

Style tags will now be extracted and loaded once per Web Template.
So, if the template is used multiple times on the same page, the style
tag will be rendered only once.

netchampfaris committed:
fix: UX

- Show route and published in list view of Web Page
- Show Type and Standard in list view of Web Template

netchampfaris committed:
fix: Move navbar styles to navbar.scss

- Add left margin to Navbar CTA

netchampfaris committed:
fix: Delete old theme files if theme is custom

netchampfaris committed:
fix: Check if file exists before generating base64

netchampfaris committed:
fix: Succinct syntax for rendering web templates

netchampfaris committed:
fix: Handle jpe files in image_to_base64

netchampfaris committed:
fix: Render web blocks using web_template.render

- web_block object need not be a WebPageBlock instance

netchampfaris committed:
fix: Default 3 columns for features section

netchampfaris committed:
fix: Conditionally render subtitle and image

netchampfaris committed:
fix: Navbar CTA undo margin in mobile view

netchampfaris committed:
fix: Better label for hero button action

netchampfaris committed:
feat: Align center in markdown template

netchampfaris committed:
fix: Fallback image alt

netchampfaris committed:
fix: Clear sitemap cache along with website cache

netchampfaris committed:
fix: match multiline scripts and styles

netchampfaris committed:
fix: Render html as text to prevent html injection

netchampfaris committed:
feat: Split section with image

- Vertical align content to top or middle
- Configurable column split

netchampfaris committed:
fix: Set type as Section

netchampfaris committed:
fix: Prevent shrink in collapsible icon

netchampfaris committed:
fix: Set max width on hero title

netchampfaris committed:
fix: Brand input styles

netchampfaris committed:
fix: Scope card styles to section-with-cards

netchampfaris committed:
Merge remote-tracking branch 'upstream/develop' into aaaa-website-enhancements-for-redesign

netchampfaris committed:
style: sider suggestions
